client/my-sites/design-preview/index.js

```diff
@@ -165,36 +168,43 @@ const DesignPreview = React.createClass( {
 			return;
 		}
 		event.preventDefault();
-		// TODO: if the href is on the current site, load the href as a preview and fetch markup for that url
 	},
 
-	getPreviewUrl( site ) {
-		if ( ! site ) {
+	getPreviewUrl() {
+		if ( ! this.props.selectedSiteUrl && ! this.props.previewUrl ) {
```

Comment:
Playing with ideas to simplify this a bit:

```diff
diff --git a/client/my-sites/design-preview/index.js b/client/my-sites/design-preview/index.js
index f14b842..29d0927 100644
--- a/client/my-sites/design-preview/index.js
+++ b/client/my-sites/design-preview/index.js
@@ -5,7 +5,6 @@ import React from 'react';
 import { connect } from 'react-redux';
 import noop from 'lodash/noop';
 import debugFactory from 'debug';
-import url from 'url';
 
 /**
  * Internal dependencies
@@ -19,6 +18,7 @@ import { updatePreviewWithChanges } from 'lib/design-preview';
 import layoutFocus from 'lib/layout-focus';
 import { getSelectedSite, getSelectedSiteId, getPreviewUrl } from 'state/ui/selectors';
 import { getSiteOption } from 'state/sites/selectors';
+import addQueryArgs from 'lib/route/add-query-args';
 
 const debug = debugFactory( 'calypso:design-preview' );
 
@@ -170,24 +170,15 @@ const DesignPreview = React.createClass( {
        },
 
        getPreviewUrl() {
-               if ( ! this.props.selectedSiteUrl && ! this.props.previewUrl ) {
-                       debug( 'no preview url and no site url were found for this site' );
-                       return null;
-               }
-               const baseUrl = this.props.previewUrl || this.props.selectedSiteUrl;
-               const parsed = url.parse( baseUrl, true );
-               delete parsed.search;
-               parsed.query.iframe = true;
-               parsed.query.theme_preview = true;
-               if ( this.props.selectedSiteNonce ) {
-                       parsed.query[ 'frame-nonce' ] = this.props.selectedSiteNonce;
-               }
-               const previewUrl = url.format( parsed ) + '&' + this.state.previewCount;
-               debug( 'using this preview url', previewUrl );
-               return previewUrl;
+               return addQueryArgs( {
+                       iframe: true,
+                       theme_preview: true,
+                       'frame-nonce': this.props.selectedSiteNonce || [],
+                       cachebust: this.state.previewCount
+               }, this.getBaseUrl() );
        },
 
-       getExternalUrl() {
+       getBaseUrl() {
                return this.props.previewUrl || this.props.selectedSiteUrl;
        },
 
@@ -203,7 +194,7 @@ const DesignPreview = React.createClass( {
                        <WebPreview
                                className={ this.props.className }
                                previewUrl={ useEndpoint ? null : this.getPreviewUrl() }
-                               externalUrl={ this.getExternalUrl() }
+                               externalUrl={ this.getBaseUrl() }
                                showExternal={ true }
                                showClose={ this.props.showClose }
                                showPreview={ this.props.showPreview }
```

The `|| []` on the frame nonce is very hacky 😄  https://github.com/nodejs/node-v0.x-archive/issues/7971